code/game/machinery/computer/arcade.dm

```diff
@@ -1075,3 +1075,272 @@
 #undef ORION_STATUS_NORMAL
 #undef ORION_STATUS_GAMEOVER
 #undef ORION_STATUS_MARKET
+
+//////////////////
+// Claw Machine //
+//////////////////
+
+/obj/machinery/computer/arcade/clawmachine
+	name = "AlliCo Grab-a-Gift"
+	desc = "Show off your arcade skills for that special someone!"
+	icon_state = "clawmachine"
+	icon_keyboard = null
+	icon_screen = null
+	circuit = /obj/item/weapon/circuitboard/arcade/clawmachine
+	prizes = list(/obj/random/plushie)
+	var/wintick = 0
+	var/winprob = 0
+	var/instructions = "Insert 1 thaler or swipe a card to play!"
+	var/gameStatus = "CLAWMACHINE_NEW"
+	var/gamepaid = 0
+	var/gameprice = 1
+	var/winscreen = ""
+
+/// Payment
+/obj/machinery/computer/arcade/clawmachine/attackby(obj/item/I as obj, mob/user as mob)
+	if(..())
+		return
+
+	var/obj/item/weapon/card/id/W = I.GetID()
+
+	if(!emagged)
+		if(gamepaid == 0 && vendor_account && !vendor_account.suspended)
+			var/paid = 0
+			var/handled = 0
+
+			if(W) //for IDs and PDAs and wallets with IDs
+				paid = pay_with_card(W,I)
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash/ewallet))
+				var/obj/item/weapon/spacecash/ewallet/C = I
+				paid = pay_with_ewallet(C)
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash))
+				var/obj/item/weapon/spacecash/C = I
+				paid = pay_with_cash(C, user)
+				handled = 1
+
+			if(paid)
+				gamepaid = 1

```

Comment:
You can just assign `gamepaid = [pay_woth_x()]` and check the value of that in this if, the paid car is redundant

Reply:
Not sure what you mean here.